### PR TITLE
[ENG-7455] Add better message on doctor timeout

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

Better error message when expo doctor times out

# How

# Test Plan

not tested

### CODE REVIEW

File: packages/build-tools/src/common/setup.ts [hunk 1]
```diff
@@ -78,6 +84,7 @@ export async function setupAsync<TJob extends Job>(ctx: BuildContext<TJob>): Pro
 async function runExpoDoctor<TJob extends Job>(ctx: BuildContext<TJob>): Promise<SpawnResult> {
   ctx.logger.info('Running "expo doctor"');
   let timeout: NodeJS.Timeout | undefined;
+  let didTimedout = false;
```
Comment (dsokal) [inline packages/build-tools/src/common/setup.ts:87]:
```suggestion
  let timedOut = false;
```

File: packages/build-tools/src/common/setup.ts [hunk 2]
```diff
@@ -68,7 +70,11 @@ export async function setupAsync<TJob extends Job>(ctx: BuildContext<TJob>): Pro
           ctx.markBuildPhaseHasWarnings();
         }
       } catch (err) {
-        ctx.logger.error({ err }, 'Command "expo doctor" failed.');
+        if (err instanceof DoctorTimeoutError) {
+          ctx.logger.error(err.message);
```
Comment (dsokal) [inline packages/build-tools/src/common/setup.ts:74]:
```suggestion
          ctx.logger.error({ err }, err.message);
```

File: packages/build-tools/src/common/setup.ts [hunk 3]
```diff
@@ -91,12 +98,18 @@ async function runExpoDoctor<TJob extends Job>(ctx: BuildContext<TJob>): Promise
...
+      didTimedout = true;
       promise.child.kill();
       ctx.reportError?.(`"expo doctor" timed out`, undefined, {
         extras: { buildId: ctx.env.EAS_BUILD_ID },
       });
     }, MAX_EXPO_DOCTOR_TIMEOUT_MS);
     return await promise;
+  } catch (err: any) {
+    if (didTimedout) {
+      throw new DoctorTimeoutError('"expo doctor" check took too long, skipping the step.');
```
Comment (dsokal) [inline packages/build-tools/src/common/setup.ts:110]:
```suggestion
      throw new DoctorTimeoutError('"expo doctor" timed out, skipping the step.');
```

**Review by dsokal — APPROVED** [review]

File: packages/build-tools/src/common/setup.ts (hunk 2 above)
Comment (wkozyra95) [reply packages/build-tools/src/common/setup.ts:74]: That would duplicate error in logs

**Pull request merged**